pkg/http/http.go

```diff
@@ -113,18 +132,35 @@ func (c *HTTPClient) setAuth(req *http.Request) {
 }
 
 func (c *HTTPClient) MakeRequest(ctx context.Context) (*http.Response, error) {
-	req, err := http.NewRequest(http.MethodGet, c.config.URL, nil)
+	method := "GET"
+	if c.config.Method != "" {
+		method = c.config.Method
+	}
+	var body []byte
+	if c.config.Method == "POST" {
+		var err error
+		if c.config.Body == "" {
+			return nil, fmt.Errorf("body is required for POST method")
```

Comment:
Payload is not mandatory for http post, this check is not required

Reply:
done